src/common/task.cc

```diff
@@ -82,6 +82,13 @@ class TaskBuilder {
   }
 
   uint8_t *Finish(int64_t *size) {
+    /* For actor tasks, other than the first, add a dependency on the dummy
+     * output of the previous task. */
+    if (!ActorID_equal(actor_id_, NIL_ACTOR_ID) && actor_counter_ > 0) {
+      ObjectID predecessor_id =
+          task_compute_return_id(actor_id_, actor_counter_ - 1);
+      NextReferenceArgument(predecessor_id);
+    }
```

Comment:
This should probably be done in Python, e.g., simply by passing the dummy object in as an argument to the task spec.

E.g., in the block

```python
            task = ray.local_scheduler.Task(
                self.task_driver_id,
                ray.local_scheduler.ObjectID(function_id.id()),
                args_for_local_scheduler,
                function_properties.num_return_vals,
                self.current_task_id,
                self.task_index,
                actor_id,
                self.actor_counters[actor_id],
                [function_properties.num_cpus, function_properties.num_gpus,
                 function_properties.num_custom_resource])
```

in worker.py.

Reply:
We could, but I did it here so that we could create an object ID that was a function of the `actor_id` and `actor_counter` values. We could do it in Python too, but we would have to duplicate or expose the C code for `task_compute_return_id`. What do you think?

Reply:
Shouldn't all of the return object IDs should already be a function of the `actor_id` and `actor_counter` values because they are computed from the `task_id` which hashes in those two values (see https://github.com/ray-project/ray/blob/master/src/common/task.cc#L57-L58)?

Reply:
Yes, but they are also a function of the rest of the task. I believe it's better to have it as a function of just `actor_id` and `actor_counter`, so that anyone can determine the object ID returned by that actor task. If want to do it just in Python, the other option would be to record the dummy objects on the caller, in addition to the current `actor_counter`.

Reply:
Let's try to make the code work regardless of how the object IDs are generated (assuming they are generated in deterministic and collision-avoiding manner). We can always bake more semantics into the IDs, but I'd prefer to avoid that unless there is a very good reason.

I agree that recording the dummy objects on the caller make sense. In fact, I thought that's what this PR was doing (but I see that it's not now).